userprofile/templates/userprofile/login.html

```diff
@@ -35,7 +58,7 @@ <h2 id="login-haka-users">
 						{% endif %}
 						<p>
 							<a class="aplus-button--secondary aplus-button--lg" role="button"
-								href="{% url 'shibboleth-login' %}?next={{ next|iriencode }}">
+								href="{% url 'haka-login' %}?target={% url 'shibboleth-login' %}">
```

Comment:
Is the query parameter (after `?`) safe without any encoding? Special characters might require encoding to be safe and standards-compliant. Could you check if it is required?

Reply:
It comes from our own source code,in urls.py and equals to "shibboleth/login/", so I thought the encoding is not needed.

Reply:
Yeah. I wasn't sure if Django includes the domain (https://plus.cs.aalto.fi) in the output of the `url` template tag. I suppose this is fine.